src/overlays/actors/ovl_En_Famos/z_en_famos.c

```diff
@@ -28,133 +66,792 @@ const ActorInit En_Famos_InitVars = {
+        this->actor.speedXZ = 0.0f;
+        EnFamos_SetupChase(this);
+    }
+}
+
+/**
+ * Summary: Looking around for player.
+ */
+void EnFamos_SetupScanForPlayer(EnFamos* this) {
+    this->stateTimer = 60;
+    this->actionFunc = EnFamos_ScanForPlayer;
+    this->actor.speedXZ = 0.0f;
+}
+
+void EnFamos_ScanForPlayer(EnFamos* this, GlobalContext* globalCtx) {
+    EnFamos_UpdateBobbingHeight(this);
+    this->stateTimer--;
+    if (EnFamos_IsPlayerSeen(this, globalCtx)) {
+        EnFamos_SetupAlert(this);
+    } else if (this->stateTimer == 0) {
+        EnFamos_SetupTurnHome(this);
+    } else {
+        // cast req
+        this->actor.shape.rot.y = (s32)(Math_SinS(this->stateTimer * 0x888) * 8192.0f) + this->actor.world.rot.y;
+    }
+}
+
+/**
+ * Summary: Famos was upside down and hit the ground headfirst. This kills the Famos.
+ */
+void EnFamos_SetupDeathSlam(EnFamos* this) {
+    this->emblemCollider.base.acFlags &= ~AC_ON;
+    this->stateTimer = 20;
+    this->actor.speedXZ = 0.0f;
+    Actor_SetColorFilter(&this->actor, 0x4000, 0xFF, false, 20);
+    this->flippedTimer = -1;
+    this->actor.world.pos.y = this->actor.floorHeight - 60.0f;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_EYEGOLE_DEAD);
+    this->actionFunc = EnFamos_DeathSlam;
+}
+
+void EnFamos_DeathSlam(EnFamos* this, GlobalContext* globalCtx) {
+    if (this->stateTimer == 17) {
+        this->collider2.base.atFlags &= ~AT_ON;
+    }
+
+    if (this->stateTimer == 0) {
+        EnFamos_SetupDeathExplosion(this);
+    } else {
+        this->stateTimer--;
+    }
+}
+
+void EnFamos_SetupDeathExplosion(EnFamos* this) {
+    this->actor.world.rot.x = 0x4000;
+    Actor_SetColorFilter(&this->actor, 0x4000, 0xFF, false, 4);
+    this->stateTimer = 25;
+    Math_Vec3f_Copy(&this->targetDest, &this->actor.world.pos);
+    this->actor.flags |= ACTOR_FLAG_10;
+    this->actionFunc = EnFamos_DeathExplosion;
+}
+
+void EnFamos_DeathExplosion(EnFamos* this, GlobalContext* globalCtx) {
+    Math_StepToF(&this->actor.speedXZ, 3.0f, 0.3f);
+    if (this->actor.colorFilterTimer == 0) {
+        Actor_SetColorFilter(&this->actor, 0x4000, 0xFF, false, 4);
+    }
+
+    this->actor.world.pos.x = randPlusMinusPoint5Scaled(5.0f) + this->targetDest.x;
+    this->actor.world.pos.z = randPlusMinusPoint5Scaled(5.0f) + this->targetDest.z;
+    if (this->stateTimer == 1) {
+        EnBom* blast = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.world.pos.x,
+                                           this->actor.world.pos.y + 40.0f, this->actor.world.pos.z, 0, 0, 0, ENBOM_0);
+        if (blast != NULL) {
+            blast->timer = 0; // instant explosion
+        }
+        this->stateTimer--;
+
+    } else if (this->stateTimer == 0) {
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos,
+                                   (0xD << 4)); // random item from droptable 0xD
```

Comment:
Disagree. `0xD0` being passed as a parameter is unreadable magic number.

Reply:
It would probably be better to document `Item_DropCollectibleRandom`, as this will probably have a bit packing macro eventually, and having the documentation split through different actors will just be messy. Perhaps adding a comment like "`0-16 << 4` designates the drop table" above `Item_DropCollectibleRandom`.

Reply:
Exactly, this is something that needs centralised documentation, not isolated comments scattered throughout the codebase.

Reply:
I agree there should be macros for this, but it is beyond the scope of this PR to add one. Until the macros are added I see no reason to remove documentation explaining what the code is doing. At the time the macro is added, it is highly likely the macros will be applied to existing code, at which time this documentation can be easily spotted as they are on the same line.

Reply:
Assuming it's not done via a script.